metrics/formatters.go

```diff
@@ -0,0 +1,237 @@
+package metrics
+
+import (
+	"encoding/json"
+	"github.com/openziti/foundation/metrics/metrics_pb"
+	"github.com/openziti/foundation/util/iomonad"
+	"io"
+	"strings"
+)
+
+type Formatter interface {
+	WriteTo(msg *metrics_pb.MetricsMessage, out io.Writer) error
+}
+
+type PlainTextFormatter struct {
+}
+
+func (PlainTextFormatter) WriteTo(msg *metrics_pb.MetricsMessage, out io.Writer) error {
+	w := iomonad.Wrap(out)
+	for name, val := range msg.IntValues {
+		w.Printf("%v: %9d\n", name, val)
+	}
+
+	for name, val := range msg.FloatValues {
+		w.Printf("%s: %v\n", name, val)
+	}
+
+	for name, val := range msg.Histograms {
+		w.Printf("histogram %s\n", name)
+		w.Printf("  count:       %9d\n", val.Count)
+		w.Printf("  min:         %9d\n", val.Min)
+		w.Printf("  max:         %9d\n", val.Max)
+		w.Printf("  mean:        %12.2f\n", val.Mean)
+		w.Printf("  stddev:      %12.2f\n", val.StdDev)
+		w.Printf("  50%%:         %12.2f\n", val.P50)
+		w.Printf("  75%%:         %12.2f\n", val.P75)
+		w.Printf("  95%%:         %12.2f\n", val.P95)
+		w.Printf("  99%%:         %12.2f\n", val.P99)
+		w.Printf("  99.9%%:       %12.2f\n", val.P999)
+		w.Printf("  99.99%%:      %12.2f\n", val.P9999)
```

Comment:
No-op for this PR, but we're probably going to want a smaller (vertically+density) output version of this for development use. We probably don't need every variant of the metric in most development scenarios.